Changing wording and syntax in Hai Revealed 1 - 4 (#7512)
endless-sky/endless-sky: data/hai/hai reveal 1 intro.txt

		has "event: pug media coverage"
	on offer
		event "hai secret leaks" 10 15

> 		log "Minor People" "Maduena Kabiru" `A councilor from Bounty in the Delta Velorum system. He is somewhat paranoid and insistent upon the Navy taking great steps to better explore the galaxy beyond human space to search for possible threats.`

Author: Alright, I'll make it senator, but I still think it's too inconsistent.

	`	"So now in our surprise segment, let me introduce you to Ronald Campbell. He once crewed on one of these big trade ships, but after seeing what he saw, he just had to come forward!"`
	`	The host stresses the word 'had' so much his voice squeaks, but the audience's attention has certainly been gained.`
	`	"But first, let's take a look at a clip."`
	# Add options to watch the clip. At least this makes it more like other events. 
	branch "hai knowledge"
		has "First Contact: Hai: offered"
	`	They show a short video featuring alien ships, not Pug, or Quarg, but something else. They are engaged in combat with each other, and fire weapons quite unlike anything produced in human space.`
	`	You raise your eyebrows. Actual footage of an unknown alien conflict doesn't seem likely.`

> 	`	When the clip ends, Campbell is encouraged by Camembert to talk about where he got the footage. Campbell is well presented - although his suit is definitely rented, because you couldn't imagine how a crewman would be able to afford a suit like that on their salary - but he's clearly anxious about what he's exposed. You can make out the beads of sweat on his brow as he considers his answer, and his neat suit crumples slightly. When Campbell speaks, it is with a paranoid manner that is plain to see even through a screen. He talks at length about an ongoing conflict that threatens to spill over and drag all of humanity back to war; a war waged by an alien species known as the 'Hai'.`

Author: Yes, they're mentioned above at line 76 and 77.

	`	The idea that there's a directly accessible alien race, at war, that has also somehow been kept a secret, may just be a ridiculous rumor. Nevertheless, it is intriguing... You listen in to collect the relevant details for future reference.` # Add an option to listen for more information, to break it up.  

> 	`	It's difficult to be sure where you should start looking, but it seems that these 'Hai' are meant to be northward, and the biggest hub at that end of the known galaxy is Sunracer.` # And the player could be told to go to Prime in Betelgeuse as well. If the player's met the guy asking for intelligent squirrels, that could be mentioned.

Author: I will do that soon.

	`	It might be time to go visit your Hai friends again.`
		accept

conversation "hai reveal republic or syndicate news"
	`You are relaxing at the spaceport while your ship refuels, when a breaking news broadcast comes on; it's not urgent enough to actually interrupt scheduled content, but urgent enough to fill the first ad break.`
	`	The severe-looking anchorwoman bores down the camera with an expression that demands attention.`
	`	"An ex-crewman employed by an itinerant trader has come forward to media in the Free Worlds about hostile aliens just beyond our borders. He claims to have brought video proof of an ongoing conflict that threatens to spill over and drag all of humanity back to war. Stay tuned for our full segment at six."`
	`	You raise your eyebrows. Actual footage of an unknown alien conflict doesn't seem likely.`

> 	`	That such news should come out so soon after the Pug smacks of a hoax. A quick glance at the clock shows you only have ten minutes to wait, so you stick around for the full segment out of curiosity.` # There should be an option to wait and see, to break the dialog up.

Author: I felt that pauses were necessary. If I'm reading, I can get lost on a page if I look away, and having those pauses allows me to know where I am.

> 	`	That such news should come out so soon after the Pug smacks of a hoax. A quick glance at the clock shows you only have ten minutes to wait, so you stick around for the full segment out of curiosity.` # There should be an option to wait and see, to break the dialog up.

Reviewer: If you can think of a way to have a choice here where there's at least some variation in the next line of text as a result of that choice, I think this is a fair place for that to happen.

> 	`	That such news should come out so soon after the Pug smacks of a hoax. A quick glance at the clock shows you only have ten minutes to wait, so you stick around for the full segment out of curiosity.` # There should be an option to wait and see, to break the dialog up.

Author: Ok, I might start on that tomorrow. 

Playing some of the next bits of the campaign now and the dialog texts are really long. That is another PR for another day though

	`	The news returns with the severe anchorwoman. It turns out that the crewman in question had been picked up by a minor media outlet on Tarazed. He had apparently traveled many weeks before deciding to seek the media in order to avoid being shut down by Republic Intelligence if he contacted them elsewhere. The man is fairly young, and appears very well presented, though he has a bit of the paranoid about him that is plain to see even through a screen.`
	branch "hai knowledge"
		has "First Contact: Hai: offered"
	`	They show a short video featuring alien ships, not Pug, or Quarg, but something else. They are engaged in combat with each other, and fire weapons quite unlike anything produced in human space.`

> 	`	It is only after this that the anchorwoman reveals that, while this was initially treated as a hoax, one of their own reporters posed a timely question to a well-respected trader, and caused quite the stir when he slipped up and confirmed the existence of the aliens known as 'Hai'.`

Reviewer 2: ```suggestion
	`	It is only after this that the anchorwoman reveals that, while this was initially treated as a hoax, one of their own reporters posed a timely question to a well-respected trader,resulting in quite the stir when he slipped up and confirmed the existence of the aliens known as 'Hai'.`
```

	choice
		# Rewrote question so it makes sense whether you met them or not
		`	"Who are the Unfettered?"`
	`	She seems surprised that you asked and responds, "Oh, they're a sort of rebel Hai group. If you don't know you should go visit Cloudfire in the Wah Ki system up north, or maybe even Darkcloak if you dare. A short visit would explain it far better than I ever could."`

conversation "What is going on in Hai space? (Met the Unfettered.)"
	`Landing in Hai space draws a much different type of attention than expected. Only recently you've begun to acclimate yourself to getting positive attention. As a war hero you're something of a minor celebrity now back in human space, but here suddenly mothers are keeping tight grips on their children, and passers-by look upon you with a hint of wariness.`
	`	You realize that the few humans who have settled here among the Hai are all wearing pink at the moment. When you catch the eye of one lady who looks friendly, you greet her to ask a few questions.`
	choice
		`	"Why does everyone seem so afraid?"`
		`	"Why are you all wearing pink?"`
			goto pink

> 	`	She shakes her head and wrinkles her nose with a look of distaste. "Some marauder came here just yesterday, posing as a trader to investigate the rumors. We thought the Hai had means of keeping them out, but this one snuck in and landed at the spaceport without any opposition. Then his crew kidnapped nearly four dozen Hai and hijacked a Geocoris before making off with their spoils. We're just in shock, I suppose. It's completely unprecedented for something like that to happen in broad daylight."`

Reviewer 2: ```suggestion
	`	She shakes her head and wrinkles her nose with a look of distaste. "Some marauder came here just yesterday, posing as a trader to investigate the rumors. We thought the Hai had means of keeping them out, but this one snuck in and landed at the spaceport without any opposition. Then his crew kidnapped nearly four dozen Hai and hijacked a Geocoris before making off with their spoils. We're just... in shock, I suppose. It's completely unprecedented for something like that to happen in broad daylight."`
```

I think this may capture the intended pacing a bit better even than the original with the incorrect comma placement.

	`	She sighs deeply and says, "Naturally they've ramped up security, but everyone is just a bit shaken and distrustful at the moment."`
	`	"Oh that's awful!" you say.`

> 	`	She nods and forces a little bit of false cheer into her voice, "This is why we're all wearing pink, it's just a little show of local solidarity.`

Reviewer 2: ```suggestion
	`	She nods and forces a little bit of false cheer into her voice, "This is why we're all wearing pink, just a little show of local solidarity.`
```

As dialogue I think this one was good as is.

		goto main
	label pink

> 	`	A sad smile crosses her face and her expression goes sombre as she explains, "A pirate marauder snuck down onto the planet recently, stole a Geocoris, and kidnapped a number of Hai, probably into slavery. The Hai are in shock, it's just completely unprecedented for something like that to happen in broad daylight. Naturally, they've ramped up security, but everyone is a bit shaken up. Us locals are wearing pink in solidarity, and so we're not mistaken for a security risk.`

Reviewer: Now that you've drawn my attention to this, I think this would be preferable:
```suggestion
	`	A sad smile crosses her face and her expression goes sombre as she explains, "A pirate marauder snuck down onto the planet recently, stole a Geocoris, and kidnapped a number of Hai, probably into slavery. The Hai are in shock, it's just completely unprecedented for something like that to happen in broad daylight. Naturally, they've ramped up security, but everyone is a bit shaken up so us locals are wearing pink in solidarity... and so we're not mistaken for a security risk.`
```

	label main

> 	`	"Unfortunately, it's taking a while for them to redistribute an appropriately resourced security force because of the Unfettered. So far they've only been able to recall those on leave and up their vigilance."`

Reviewer: As before

		event "news: hai video clip published"
		conversation "hai reveal republic or syndicate news"
	on visit

> 		dialog "You should first visit Allhome, where most of the humans are in Hai space, to see how things are going before you get more heavily involved."

Reviewer 2: ```suggestion
		dialog "You should first visit Allhome, where most of the humans in Hai space congregate, to see how things are going before you get more heavily involved."
```

"Live" would work as well, if not better, if it's accurate, but I can't recall if that's so. "Are" is just so poorly vague anyway, but also repeated in such a short time.

Author: Allhome should be the Hai planet with the most humans, because there are supposed to be small cities with populated by humans near the spaceport, and it was settled jointly by humans and Hai. Greenwater only has "scattered" human fishing villages but still a "sizable" human population.

> 		dialog "You should first visit Allhome, where most of the humans in Hai space live, to see how things are going before you get more heavily involved."

Reviewer: Different meaning:
The first one means that Allhome simply has the largest population of humans.
Your change means that Allhome explicitly has >50% of humans in Hai space.

Those are different and it's not a change I like.

		event "news: hai video clip published"
		conversation "hai reveal free worlds news"
	on visit

> 		dialog "You should first visit Allhome, where most of the humans are in Hai space, to see how things are going before you get more heavily involved."

Reviewer 2: ```suggestion
		dialog "You should first visit Allhome, where most of the humans in Hai space congregate, to see how things are going before you get more heavily involved."
```

As above.

> 		dialog "You should first visit Allhome, where most of the humans in Hai space live, to see how things are going before you get more heavily involved."

Reviewer: as before

mission "ZZ Event Hai Secret Leaks: RS"
	landing
	name "Investigate the Hai"

> 	description "There have been reports of a secret, violent alien race that traders have known about for ages. The rumours seem unsubstantiated, but you should go and find out what's really going on."

Reviewer: Descriptions overrun the text box if they're too long. I wouldn't extend this unless you first check that it fits in-game.

Author: I got rid of "The rumours seem unsubstantiated"; and anyway it's an example of telling, not showing.

		conversation
			`Arriving at the only place you can think of where you feel sure you can find out more information, you find that <origin>, like most places, is abuzz with the recent news. Avoiding the many large, cheap, and boisterous establishments that service the many crews and workers of the Megaparsec corporate machinery here, you make your way to one of the marginally more up-market places, cunningly marketed as a merchant-only dive.`
			`	Since the story first broke, there has been an endless cycle of contradictory reporting on the matter plastered across most screens in major thoroughfares. Here, however, it is remarkably absent, with only one of the screens on display covering the "violent alien race" at all - running stats on how many different versions of the story are now being told.`

> 			`	Casting your eyes around the room, you pick out a merchant who looks like she's been around the block a few times, and looks like she could pick up most people and break them over her knee. She's especially conspicuous with her threatening glare that you watch her give a younger captain, who left a table talking about the Hai to approach her... briefly. After the captain is briskly sent away, you make your way over. Her glare falters in a flash of recognition before subsiding to a reserved scowl.`

Reviewer 2: ```suggestion
			`	Casting your eyes around the room, you pick out a merchant who looks like she's been around the block a few times and also like she could pick up and break most people over her knee. Especially conspicuous is the threatening glare that you watch her give a younger captain when they leave a table talking about the Hai to approach her... briefly. After the captain is briskly brushed aside, you make your way over. Her glare falters in a flash of recognition before subsiding to a reserved scowl.`
```

This is still more than a bit awkward, but headed in the right direction.

Author: Yes, it will have to be looked at later, but it's been changed to your suggestion for now.

> 			`	She waits until you're directly next to her table. "The famous Captain <last>. Which ancient Sumerian deity did I offend to be granted the gift of your attention?"`

Reviewer 2: ```suggestion
			`	She waits until you're directly next to her table before acknowledging you further. "The famous Captain <last>. Which ancient Sumerian deity did I offend to be granted the gift of your attention?"`
```

Let's really drive home her behavior.

			`	Since the story first broke, there has been an endless cycle of contradictory reporting on the matter plastered across most screens in major thoroughfares. Here, however, it is remarkably absent, with only one of the screens on display covering the "violent alien race" at all - running stats on how many different versions of the story are now being told.`
			`	Casting your eyes around the room, you pick out a merchant who looks like she's been around the block a few times and also like she could pick up and break most people over her knee. Especially conspicuous is the threatening glare that you watch her give a younger captain, when they leave a table talking about the Hai to approach her... briefly. After the captain is briskly brushed aside, you make your way over. Her glare falters in a flash of recognition before subsiding to a reserved scowl.`

> 			# choice to approach her?

Reviewer: There's not really a choice though unless we implement someone different to ask. Your choices at present would be approach her, or abandon the story.

	to offer
		has "ZZ Hai Tour [1] Stonebreak: done"
	on offer

> 		conversation # The dialog should be changed to give some new information, instead of repeating the planet's description; or something else. 

Reviewer: As mentioned the whole first contact + tour will be reworked as a separate project. The comments are simply unnecessary. It's already on the To-Do list

				`	"That seems practical."`
				`	"I don't really have an option to refuse, do I?"`
			`	"Fantastic, you understand the situation," she says, though her tone is much more businesslike than the wording would imply. "In that case, if you could please take this to Earth and communicate with your various governments, then we can begin a dialogue on how best to deal with this current scenario."`

> 			`	She hands you an officially branded box containing what appears to be a diplomatic ID. The text on the ID appears to identify you in a few scripts you've seen in human space and definitely recognize as human, but it also has another script you've never seen before that must be in Hai. You wonder for a brief moment how they were able to gain this information on you ahead of time.` 

Reviewer: It has occurred to me in reviewing certain things through that this ID wouldn't have "script" on it. It would either be some kind of badge/seal of a specific unforgeable design that simply denotes you as an emissary of the Hai, or it would be digital of some kind - potentially holographic. Probably keyed to recognise spoken language and display its information in that language (if it's one that it recognises).

			choice
				`	"That seems practical."`
				`	"I don't really have an option to refuse, do I?"`

> 			`	"Fantastic, you understand the situation," she says, though her tone is much more businesslike than the wording would imply. "In that case, if you could please take this to Earth and communicate with your various governments, then we can begin a dialogue on how best to deal with this current scenario."`

Reviewer: There's a sentence in the codespell file that is used to make this line pass checks.
That will need to be updated to match this line exactly.

This is because for this particular line "dialogue" is absolutely the correct term, but in general the game needs to check that "dialog" is used correctly in all the usual places it is.

	passengers 4
	on offer
		event "pirates respond to leaks"

> 		log "People" "Yang Xilin" `Deputy Minister Yang works in the Foreign Affairs department of the Republic government. He is a pleasant and astute man. Handling the Hai secrecy crisis has become his job while the majority of the foreign ministry remains engaged with the fallout from the Free Worlds and recent peace.`

Author: Yang is his last name, this is how Chinese names work, and why you usually hear for Xi Jinping, "President Xi", instead of "President Jinping". Jinping is the first name. I commented earlier: 

> By the way, Chinese names go surname:first name, so Xilin's proper full name would be Yang Xilin. I don't think this would have changed in a thousand years between now and Endless Sky. 

Deputy Minister Yang is still correct, and calling him Xilin is also correct. There should be no conflict with his name in the rest of the story.

> 		log "People" "Yang Xilin" `Deputy Minister Yang works in the Foreign Affairs department of the Republic government. He is a pleasant and astute man. Handling the Hai secrecy crisis has become his job while the majority of the foreign ministry remains engaged with the fallout from the Free Worlds and recent peace.`

Reviewer 3: I know, and maybe it's a regional thing, but most of the Chinese I've known (english/spanish/french speaking country here) reverse it, using the western convention since westerners will institutionally call them by their first name. If people call everyone in your family by the same name, it gets confusing when talking to strangers (mother, father, sister, and dog are all Zhang). The humans in Endless Sky will probably have the same issue since all of the characters thus far have used the First Last ordering.

> 		log "People" "Yang Xilin" `Deputy Minister Yang works in the Foreign Affairs department of the Republic government. He is a pleasant and astute man. Handling the Hai secrecy crisis has become his job while the majority of the foreign ministry remains engaged with the fallout from the Free Worlds and recent peace.`

Author: My tone might have been too harsh, haha.

This may be a regional thing. The city I live in (not close to China, but around 5-10% population is Chinese), and the family I have, will always use last:first. If they have to, they'll also have english names, but among themselves they will always use their Chinese name. Otherwise, people seem to get around it fine.

> 		log "People" "Yang Xilin" `Deputy Minister Yang works in the Foreign Affairs department of the Republic government. He is a pleasant and astute man. Handling the Hai secrecy crisis has become his job while the majority of the foreign ministry remains engaged with the fallout from the Free Worlds and recent peace.`

Reviewer: Yeah it's very situationally dependent on how integrated they are with other cultures I find.

This particular name is based on a colleague of mine who uses their name in the way it's used here.
The subtle implication being that he isn't particularly close to his roots.

> 		log "People" "Yang Xilin" `Deputy Minister Yang works in the Foreign Affairs department of the Republic government. He is a pleasant and astute man. Handling the Hai secrecy crisis has become his job while the majority of the foreign ministry remains engaged with the fallout from the Free Worlds and recent peace.`

Reviewer 2: With that being the explicit intent, I think reverting it is the best option.

Without that though, I do think the change was the best course.

Another thing I'd like to see hashed out (later, in a separate PR) then, though, is some kind of passing reference to reinforce that implication.

			label next
			`	"Now, allow me to introduce you to Terry," Remington says while gesturing to the woman behind him.`
			`	"Terry Adrianopoulos. Pleasure to meet you," she says with a broad smile and a slight bow. The young woman goes from calculating to gregarious in the blink of an eye as your attention shifts to her. She seems at once pleasant and approachable while at the same time being slightly off-putting and uncanny. It's an eerie feeling.`

> 			`	"Terry will be accompanying me as... well, let's call her my secretary," Remington says with a sly smile. Terry seems none too pleased to hear that, but she doesn't correct him. "I have men who can hold the fort here while I'm gone, and I really do need to get some fresh air, so the two of us will be joining you. There is an internal political situation going on within the Syndicate that has been causing me quite the headache recently," he says. You think that he may have glanced at Terry as he said that, but you aren't sure if it was just incidental or indicative of something. "Nothing I can discuss in detail, unfortunately, but something I'll certainly enjoy taking a break from.`

Author: Remington is speaking here, or else this needs to be changed so we know that it's Terry who's speaking. I took it out because Remington is acting casual here, but I can add in something like "I'm not authorised to talk to you about it", or "the details will bore you, so I won't bother" to make it more clear.

> 			`	"Terry will be accompanying me as... well, let's call her my secretary," Remington says with a sly smile. Terry seems none too pleased to hear that, but she doesn't correct him. "I have men who can hold the fort here while I'm gone, and I really do need to get some fresh air, so the two of us will be joining you. There is an internal political situation going on within the Syndicate that has been causing me quite the headache recently," he says. You think that he may have glanced at Terry as he said that, but you aren't sure if it was just incidental or indicative of something. "Nothing I can discuss in detail, unfortunately, but something I'll certainly enjoy taking a break from.`

Reviewer: It is Remington speaking, but specifically calling out the NDA's is functionally just an example of corporate humour in this case. Most captains would probably have encountered an NDA if they've ever dealt with the Syndicate.
if the player doesn't get it.. well that's part of their character then.

		has "Hai Leaks Response 3: done"
	on offer
		conversation
			`When you land on Allhome, you are greeted by the quartermaster of the local Hai security forces. You're not an expert at judging Hai ages, but he appears to be a little younger than expected.`
			`	"Captain <last>! I heard you were coming to assist."`
			choice
				`	"Time to rout some pirates, baby!"`
				`	"I'm here to reinforce you. What can I do?"`
			`	"Uh well, yes. I presume you saw the pirates on your way in? They've been appearing en masse and overwhelming security forces to raid technology and goods from the Hai traders and their security. Some of them have even been sneaking down onto the planet, capturing Hai, and taking them away."`
			choice
				`	"I have heard about the kidnapping."`

> 			`	He nods. "Unfortunately, it's taking a while for them to redistribute an appropriately resourced defense force, because of the Unfettered. It's bad enough that we've had to reinstate these old Ladybug models to bolster our forces here."`

Reviewer 2: ```suggestion
			`	He nods. "Unfortunately, it's taking a while for them to redistribute an appropriately resourced defense force, because of the Unfettered. It's bad enough that we've had to bring these old Ladybug models out of mothballs to bolster our forces here."`
```

I like the "the"->"these," change but think the less formal phrasing better demonstrates the quartermaster's frustration.

		has "Hai Leaks Response 3: done"
	on offer
		conversation
			`When you land on Allhome, you are greeted by the quartermaster of the local Hai security forces. You're not an expert at judging Hai ages, but he appears to be a little younger than expected.`
			`	"Captain <last>! I heard you were coming to assist."`
			choice
				`	"Time to rout some pirates, baby!"`
				`	"I'm here to reinforce you. What can I do?"`
			`	"Uh well, yes. I presume you saw the pirates on your way in? They've been appearing en masse and overwhelming security forces to raid technology and goods from the Hai traders and their security. Some of them have even been sneaking down onto the planet, capturing Hai, and taking them away."`
			choice
				`	"I have heard about the kidnapping."`

> 			`	He nods. "Unfortunately, it's taking a while for them to redistribute an appropriately resourced defense force, because of the Unfettered. It's bad enough that we've had to bring these old Ladybug models out of mothballs to bolster our forces here."`

Reviewer: Not a fan of either of the changes here.
